compiler/rustc_trait_selection/src/traits/wf.rs

```diff
@@ -731,6 +731,11 @@ impl<'a, 'tcx> WfPredicates<'a, 'tcx> {
                     }
                 }
 
+                ty::Alias(ty::Weak, ty::AliasTy { def_id, substs, .. }) => {
```

Comment:
Are aliases checked for well-formedness under their own param-envs? Otherwise this is possibly subtly wrong.

Reply:
yes they are.